File: src/main/java/com/moira/itda/global/exception/GlobalExceptionHandler.java
package com.moira.itda.global.exception;

import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.time.ZonedDateTime;

@RestControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(ItdaException.class)
    public ErrorResponse handleItdaException(ItdaException e) {
        return ErrorResponse.builder()
                .errorCode(e.getErrorCode())
                .message(e.getMessage())
                .time(ZonedDateTime.now())
                .build();
    }
}

File: src/main/kotlin/com/moira/itda/global/exception/GlobalExceptionHandler.kt
package com.moira.itda.global.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.multipart.MaxUploadSizeExceededException
import java.time.ZonedDateTime

@RestControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(ItdaException::class)
    fun handleItdaException(e: ItdaException): ResponseEntity<ErrorResponse> {
        val errorResponse = ErrorResponse(
            errorCode = e.errorCode,
            message = e.message ?: ErrorCode.INTERNAL_SERVER_ERROR.message,
            time = ZonedDateTime.now()
        )

        return ResponseEntity.status(e.errorCode.httpStatus).body(errorResponse)
    }

    @ExceptionHandler(MaxUploadSizeExceededException::class)
    fun handleMaxUploadSizeException(e: MaxUploadSizeExceededException): ResponseEntity<ErrorResponse> {
        val errorCode = ErrorCode.MAX_UPLOAD_SIZE_EXCEEDED
        val errorResponse = ErrorResponse(
            errorCode = errorCode,
            message = errorCode.message,
            time = ZonedDateTime.now()
        )

        return ResponseEntity.status(errorCode.httpStatus).body(errorResponse)
    }
}